### PR TITLE
Cherrypicking fixes from upstream

### PR DESCRIPTION
**Pull request by Michiyamenotehifunana**
## About The Pull Request

NO MORE GIANT REVOLVERS AND AEGS WHEN YOU PUT THEM IN THE SUIT SLOT

also bayonets work now 

This PR finishes off the optimization ports from upstream. Several other fix PRs are included as well, including one that changes all instances of "station" to "ship". Crossbow crafting has also been fully integrated into the crafting system instead of using bay's weird thing.

## Why It's Good For The Game
lag bad fix good

## Changelog
```changelog Toriate
fix: Ported a ton of things from upstream including a fix for guns not being able to have bayonets and certain guns having giant onsuit sprites.
```

### CODE REVIEW

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -61,6 +61,7 @@
 	icon_state = "emag"
 	item_state = "card-id"
 	origin_tech = list(TECH_MAGNET = 2, TECH_COVERT = 2)
+	matter = list(MATERIAL_SILVER = 1, MATERIAL_PLASTIC = 1)
```
[the remark this answers is not in the document]
Comment (Jamini) [reply code/game/objects/items/weapons/cards_ids.dm:64]: OH WAIT. IT's an emag! NEVER MIND!

File: code/modules/mining/mineral_effect.dm [hunk 2]
```diff
@@ -15,7 +15,7 @@
 	ore_key = M.name
 	SetIconState("rock_[ore_key]")
 	var/turf/T = get_turf(src)
-	T.overlays += image('icons/obj/mining.dmi', "rock_[ore_key]", dir = 1)	// OCCULUS EDIT: Show ores through walls when using mesons
+	T.overlays += image('icons/obj/mining.dmi', "rock_[ore_key]", dir = 1)
```
Comment (Jamini) [inline code/modules/mining/mineral_effect.dm:18]: THEY TOOK OUR PORT.

REEEEEE

**Review by Jamini — APPROVED** [review]
Seems overall fine. I'm glad Eris is nabbing some of our fixes.

File: code/modules/mining/mineral_effect.dm (hunk 2 above)
Comment (Zenitheevee) [reply code/modules/mining/mineral_effect.dm:18]: "We did a smarts and they stolens it"

Comment (drexample) [reply code/modules/mining/mineral_effect.dm:18]: bwehehehehehehe

**Pull request merged**